Week6/Day2/Blahgger/Blahgger/Controllers/AccountController.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Blahgger.Data;
+using Blahgger.Models;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using System.Web.Security;
+
+namespace Blahgger.Controllers
+{
+    public class AccountController : Controller
+    {
+        string connectionString = ConfigurationManager.ConnectionStrings["BlogDatabase"].ConnectionString;
+
+        [AllowAnonymous]
+        public ActionResult Login()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult Login(LoginUser user)
+        {
+            if (ModelState.IsValidField("Email") && ModelState.IsValidField("Password"))
+            {
+                // TODO get the user record from the database
+
+
+                using (SqlConnection connection = new SqlConnection(connectionString))
+                {
+                    connection.Open();
+                    string sql = @"
+                      SELECT Id, Email, HashedPassword
+                      FROM [User]
+                      Where Email = @Email
+                    ";
+                    SqlCommand command = new SqlCommand(sql, connection);
+                    command.Parameters.AddWithValue("@Email", user.Email);
+
+                    SqlDataReader reader = command.ExecuteReader();
+                    if (reader.Read())
+                    {
+                        if (!Hashing.ValidatePassword(user.Password, reader.GetString(2)))
+                        {
+                            ModelState.AddModelError("", "Login failed.");
+                        }
```

Comment:
With a little refactoring of your `if` statements, you can combine this `AddModelError` method call with the one below it.